diff -Nuar soci-4.0.3/tests/catch.hpp soci-4.0.3-new/tests/catch.hpp
--- soci-4.0.3/tests/catch.hpp	2022-02-11 03:13:13.000000000 +0800
+++ soci-4.0.3-new/tests/catch.hpp	2024-06-14 15:42:50.652732100 +0800
@@ -7166,18 +7166,19 @@
 #include <set>
 #include <sstream>
 #include <algorithm>
+#include <random>
 
 namespace Catch {
 
     struct RandomNumberGenerator {
-        typedef std::ptrdiff_t result_type;
+        using generator_type = std::mt19937_64;
 
-        result_type operator()( result_type n ) const { return std::rand() % n; }
+        using result_type = generator_type::result_type;
 
 #ifdef CATCH_CONFIG_CPP11_SHUFFLE
-        static constexpr result_type min() { return 0; }
-        static constexpr result_type max() { return 1000000; }
-        result_type operator()() const { return std::rand() % max(); }
+        result_type operator()() const {return std::rand() % max();};
+        static constexpr result_type min() { return generator_type::min(); }
+        static constexpr result_type max() { return generator_type::max(); }
 #endif
         template<typename V>
         static void shuffle( V& vector ) {
