diff -Naur ggml-0.9.4.orig/tests/test-opt.cpp ggml-0.9.4/tests/test-opt.cpp
--- ggml-0.9.4.orig/tests/test-opt.cpp
+++ ggml-0.9.4/tests/test-opt.cpp
@@ -98,7 +98,7 @@
     struct ggml_context * ctx_compute;
     {
         struct ggml_init_params params = {
-            /*.mem_size   =*/ (2*ndata + 2)*ggml_tensor_overhead(),
+            /*.mem_size   =*/ static_cast<size_t>((2*ndata + 2)*ggml_tensor_overhead()),
             /*.mem_buffer =*/ nullptr,
             /*.no_alloc   =*/ true,
         };
diff -Naur ggml-0.9.4.orig/examples/sam/sam.cpp ggml-0.9.4/examples/sam/sam.cpp
--- ggml-0.9.4.orig/examples/sam/sam.cpp
+++ ggml-0.9.4/examples/sam/sam.cpp
@@ -1623,4 +1623,4 @@
     tokens = ggml_new_tensor_3d(ctx0, GGML_TYPE_F32, dec.iou_token_w->ne[0], dec.iou_token_w->ne[1] + dec.mask_tokens_w->ne[1] + sparse->ne[1], sparse->ne[2]);

-        const size_t offsets[3] = { 0, dec.iou_token_w->ne[1]*tokens->nb[1], dec.iou_token_w->ne[1]*tokens->nb[1] + dec.mask_tokens_w->ne[1]*tokens->nb[1] };
+        const size_t offsets[3] = { 0, static_cast<size_t>(dec.iou_token_w->ne[1]*tokens->nb[1]), static_cast<size_t>(dec.iou_token_w->ne[1]*tokens->nb[1] + dec.mask_tokens_w->ne[1]*tokens->nb[1]) };
         ggml_build_forward_expand(gf, ggml_cpy(ctx0, dec.iou_token_w,   ggml_view_2d(ctx0, tokens, tokens->ne[0], dec.iou_token_w->ne[1],   tokens->nb[1], offsets[0])));