--- WebKit-webkitgtk-2.41.90/Tools/TestWebKitAPI/Tests/WTF/MathExtras.cpp	2023-08-10 13:57:48.000000000 +0800
+++ WebKit-webkitgtk-2.41.90_patch/Tools/TestWebKitAPI/Tests/WTF/MathExtras.cpp	2024-04-26 10:12:45.399006093 +0800
@@ -361,9 +361,9 @@
 {
     testClampSameSignIntegers<char, char>();
     testClampSameSignIntegers<unsigned char, unsigned char>();
-    testClampSameSignIntegers<char, int32_t>();
+    testClampSameSignIntegers<signed char, int32_t>();
     testClampSameSignIntegers<unsigned char, uint32_t>();
-    testClampSameSignIntegers<char, int64_t>();
+    testClampSameSignIntegers<signed char, int64_t>();
     testClampSameSignIntegers<unsigned char, uint64_t>();
 
     testClampSameSignIntegers<int32_t, int32_t>();
@@ -389,8 +389,8 @@
 
 TEST(WTF, clampUnsignedToSigned)
 {
-    testClampUnsignedToSigned<char, unsigned char>();
-    testClampUnsignedToSigned<char, uint32_t>();
+    testClampUnsignedToSigned<signed char, unsigned char>();
+    testClampUnsignedToSigned<signed char, uint32_t>();
     testClampUnsignedToSigned<int32_t, uint32_t>();
     testClampUnsignedToSigned<int64_t, uint64_t>();
     testClampUnsignedToSigned<int32_t, uint64_t>();
@@ -421,7 +421,7 @@
 
 TEST(WTF, clampSignedToUnsigned)
 {
-    testClampSignedToUnsigned<unsigned char, char>();
+    testClampSignedToUnsigned<unsigned char, signed char>();
     testClampSignedToUnsigned<unsigned char, int32_t>();
     testClampSignedToUnsigned<uint32_t, int32_t>();
     testClampSignedToUnsigned<uint64_t, int64_t>();
