diff --git openh264-2.6.0/Makefile openh264-2.6.0/Makefile
index 290d4b27..593b9796 100644
--- openh264-2.6.0/Makefile
+++ openh264-2.6.0/Makefile
@@ -189,7 +189,7 @@ gmp-bootstrap:
 	cd gmp-api && git fetch origin && git checkout $(GMP_API_BRANCH)
 
 gtest-bootstrap:
-	if [ ! -d gtest ] ; then git clone https://github.com/google/googletest.git gtest && \
+	if [ ! -d gtest ] ; then git clone https://gitee.com/mirrors/googletest.git gtest && \
        cd gtest && git checkout -b $(GTEST_VER) $(GTEST_VER) ; fi
 
 ifeq ($(HAVE_GTEST),Yes)
