diff -Nura clip2tri-master/CMakeLists.txt clip2tri-master_patch/CMakeLists.txt
--- clip2tri-master/CMakeLists.txt	2014-10-30 17:11:28.000000000 -0700
+++ clip2tri-master_patch/CMakeLists.txt	2023-09-26 20:05:46.554289600 -0700
@@ -26,6 +26,9 @@
 endif(HAVE_M_LIB)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/clip2tri)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/clipper)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/poly2tri)
 
 if(NOT DEFINED BIN_DIR)
   set(BIN_DIR bin)
@@ -61,9 +64,27 @@
     ARCHIVE DESTINATION ${LIB_DIR})
 endif(BUILD_STATIC_LIBS)
 
+add_executable(test test.cpp)
+target_link_libraries(test PRIVATE clip2tri)
+if(OHOS)
+target_compile_options(test PRIVATE -Wno-format-security)
+endif()
+list(APPEND TARGET_INSTALL_ELEMENT test)
+
+enable_testing()
+add_test(NAME test_clip2tri COMMAND test)
 
 # Install header files
-install(DIRECTORY clip2tri/ 
+file(GLOB CLIP2TRI_HEAD clip2tri/clip2tri.h)
+install(
+	FILES ${CLIP2TRI_HEAD}
     DESTINATION include/clip2tri
-    FILES_MATCHING PATTERN "*.h"
 )
+
+file(GLOB CLIPPER_HEAD clipper/clipper.hpp)
+install(
+	FILES ${CLIPPER_HEAD}
+    DESTINATION include/clipper
+)
+
+
diff -Nura clip2tri-master/test.cpp clip2tri-master_patch/test.cpp
--- clip2tri-master/test.cpp	2014-10-30 17:11:28.000000000 -0700
+++ clip2tri-master_patch/test.cpp	2023-09-21 02:23:00.598086266 -0700
@@ -17,6 +17,19 @@
 
    vector<Point> boundingPolygon;
 
+   Point p1 = {10, 60};
+   Point p2 = {20, 60};
+
+   vector<Point> newData = {p1,p2};
+   inputPolygons.push_back(newData);
+
+   boundingPolygon.push_back({1,2});
+   boundingPolygon.push_back({3,4});
+
    clip2tri clip2tri;
    clip2tri.triangulate(inputPolygons, outputTriangles, boundingPolygon);
+
+   for (const auto& point : outputTriangles) {
+        printf("(%2f, %2f)\n", point.x, point.y);
+    }
 }
