diff -urN spirv-reflect-sdk-1.3.204.0_old/CMakeLists.txt spirv-reflect-sdk-1.3.204.0/CMakeLists.txt
--- spirv-reflect-sdk-1.3.204.0_old/CMakeLists.txt	2024-10-07 23:12:59.000000000 +0800
+++ spirv-reflect-sdk-1.3.204.0/CMakeLists.txt	2024-10-07 23:30:34.572135368 +0800
@@ -6,7 +6,7 @@
 OPTION(SPIRV_REFLECT_EXAMPLES       "Build stripper examples" ON)
 OPTION(SPIRV_REFLECT_STRIPPER       "Build stripper utility" OFF)
 OPTION(SPIRV_REFLECT_STATIC_LIB     "Build a SPIRV-Reflect static library" OFF)
-OPTION(SPIRV_REFLECT_BUILD_TESTS    "Build the SPIRV-Reflect test suite" OFF)
+OPTION(SPIRV_REFLECT_BUILD_TESTS    "Build the SPIRV-Reflect test suite" ON)
 OPTION(SPIRV_REFLECT_ENABLE_ASSERTS "Enable asserts for debugging" OFF)
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
@@ -55,7 +55,7 @@
     SET(BUILD_GTEST ON CACHE BOOL "Builds the googletest subproject" FORCE)
     SET(INSTALL_GTEST OFF CACHE BOOL "Enable installation of googletest. (Projects embedding googletest generally turn this OFF.)" FORCE)
     SET(gtest_force_shared_crt ON CACHE BOOL "Enable DLL for C runtime libraries on Windows" FORCE)
-
+    enable_testing()  # 启用测试
     add_subdirectory(third_party/googletest)
 	add_executable(test-spirv-reflect ${CMAKE_CURRENT_SOURCE_DIR}/tests/test-spirv-reflect.cpp
                                       ${CMAKE_CURRENT_SOURCE_DIR}/spirv_reflect.h
@@ -72,6 +72,8 @@
     add_custom_command(TARGET test-spirv-reflect POST_BUILD
                        COMMAND ${CMAKE_COMMAND} -E copy_directory
                        ${CMAKE_CURRENT_SOURCE_DIR}/tests ${CMAKE_CURRENT_BINARY_DIR}/tests)
+    # 添加测试到 CTest
+    add_test(NAME Spirv_Refle_ctTest COMMAND test-spirv-reflect)
 endif()
 
 if(SPIRV_REFLECT_STATIC_LIB)
