--- SDL-release-2.0.9_old/src/thread/pthread/SDL_systhread.c	2018-10-31 01:11:02.000000000 +0800
+++ SDL-release-2.0.9/src/thread/pthread/SDL_systhread.c	2024-07-29 01:41:59.000000000 +0800
@@ -173,7 +173,7 @@
     /* Allow ourselves to be asynchronously cancelled */
     {
         int oldstate;
-        pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, &oldstate);
+        // pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, &oldstate);
     }
 #endif
 }
