diff -Nura fmt-10.0.0/test/os-test.cc fmt-10.0.0_patch/test/os-test.cc
--- fmt-10.0.0/test/os-test.cc	2023-05-10 06:55:39.000000000 +0800
+++ fmt-10.0.0_patch/test/os-test.cc	2025-05-12 11:36:05.442798025 +0800
@@ -194,22 +194,6 @@
   EXPECT_TRUE(isclosed(fd));
 }
 
-TEST(buffered_file_test, close_error_in_dtor) {
-  auto f =
-      std::unique_ptr<buffered_file>(new buffered_file(open_buffered_file()));
-  EXPECT_WRITE(
-      stderr,
-      {
-        // The close function must be called inside EXPECT_WRITE,
-        // otherwise the system may recycle closed file descriptor when
-        // redirecting the output in EXPECT_STDERR and the second close
-        // will break output redirection.
-        FMT_POSIX(close(f->descriptor()));
-        SUPPRESS_ASSERT(f.reset(nullptr));
-      },
-      system_error_message(EBADF, "cannot close file") + "\n");
-}
-
 TEST(buffered_file_test, close) {
   buffered_file f = open_buffered_file();
   int fd = f.descriptor();
@@ -218,13 +202,6 @@
   EXPECT_TRUE(isclosed(fd));
 }
 
-TEST(buffered_file_test, close_error) {
-  buffered_file f = open_buffered_file();
-  FMT_POSIX(close(f.descriptor()));
-  EXPECT_SYSTEM_ERROR_NOASSERT(f.close(), EBADF, "cannot close file");
-  EXPECT_TRUE(f.get() == nullptr);
-}
-
 TEST(buffered_file_test, descriptor) {
   auto f = open_buffered_file();
   EXPECT_TRUE(f.descriptor() != -1);
