--- a/mz_os_posix.c
+++ b/mz_os_posix.c
@@ -62,7 +62,11 @@ char *mz_os_utf8_string_create(const char *string, int32_t encoding) {
 
     if (encoding == MZ_ENCODING_UTF8)
         from_encoding = "UTF-8";
+#ifdef __OHOS__
+    else if (encoding == MZ_ENCODING_CODEPAGE_936)
+        from_encoding = "GB18030";
+#endif
     else {
         snprintf(string_encoding, sizeof(string_encoding), "CP%03" PRId32, encoding);
         from_encoding = string_encoding;
     }
